pkg/ctl/utils/associate_iam_oidc_provider.go

```diff
@@ -22,6 +22,9 @@ func associateIAMOIDCProviderCmd(cmd *cmdutils.Cmd) {
 	}
 
 	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+
+		fs.StringSliceVar(&cfg.IAM.OIDCClientIDList, "oidc-client-id-list", []string{}, "List of Client ID for OIDC provider (audience)")
```

Comment:
If a user was to provide a `ClusterConfig` with some `OIDCClientIDList` values, then providing some via the CLI flag would override these silently. 

1. Values provided via this flag should probably be stored in a different temporary slice,
2. This slice should then be passed to [`NewUtilsAssociateIAMOIDCProviderLoader`](https://github.com/weaveworks/eksctl/blob/f9e384a6efaa120a25624049c79053c560d413a5/pkg/ctl/cmdutils/configfile.go#L438).

Then, we have a choice:

- A. either fail if values are provided using _both_ CLI flags and `ClusterConfig`, in which case we need something like:
  ```go
  	l.flagsIncompatibleWithConfigFile.Insert(
  		"oidc-client-id-list",
  	)
  ```
  This is what `eksctl` typically does at the moment, for better or worse, but at least this approach would be consistent with the current behaviour.
  And if no values are in the `ClusterConfig` set it to the provided slice.
- B. merge the two lists of IDs, de-duplicate them (e.g. using a set), and then set the final result to `cfg.IAM.OIDCClientIDList` for it to be used by `eksctl`.

Disclaimer: the way these two sources of input are used together is currently, arguably, confusing, and something which we would like to decouple & make clearer.

Reply:
I guess the second option of merging and de-duplicating make sense.